### PR TITLE
Implement dataclasses needed for modules to give/take items

### PR DESCRIPTION
**Pull request by Levi-Lesches**
Notable changes:
- Rename `ModuleYieldInfo` --> `ModuleHarvestYield`
- Adds some `related_name`'s and `limit_choices_to` to the module dataclasses
- `ModuleOwnerYield`: Items the module owner gets when clicked (separate from click bonus when harvesting)
- Rename `ArcadePrize` --> `ModuleGuestYield` and used in general cases when clickers get an item from a module
- `ModuleMessage`: allows modules to send messages to users in the owner's friendlist when clicked. 

Tested, the merge works without erasing any data. The data needed to properly populate `ModuleGuestYield`, `ModuleOwnerYield`, and `ModuleMessage` are not in the database yet and need to be extracted from the XML by using the newly-updated `import_mln_xml` script.

### CODE REVIEW

File: mln/management/commands/import_mln_xml.py [hunk 1]
```diff
@@ -72,39 +76,61 @@ def handle(self, *args, **options):
...
+				for guest_yield in (yield_elem.find("guestYield") or []): 
+					probability = int(guest_yield.get("successRate") or "100")
+					item_id = int(guest_yield.get("itemID"))
+					qty = int(guest_yield.get("qty"))
+					if not ModuleGuestYield.objects.filter(module_id=id).exists():
+						t[ModuleGuestYield].append(ModuleGuestYield(module_id=id, item_id=item_id, qty=qty, probability=probability))
+
+				# Owner yield info --> ModuleOwnerYield
+				for owner_yield in (yield_elem.find("ownerYield") or []): 
+					probability = int(owner_yield.get("success") or "100")
```
[the remark this answers is not in the document]
Comment (Levi-Lesches) [reply mln/management/commands/import_mln_xml.py:105]: Hm, not sure why I included it -- modules that have a chance at yielding an item for the owner or guest aren't in the XML, but are in the database. I'll set it to be 100 by default.

**Comment by Levi-Lesches** [comment]
Okay, sorry for the long hiatus, but I fixed all your comments. I was confused at first because `ignore_conflicts` added duplicate entries and a LOT of new errors, but by adding in a new unique constraint, it works now! Caveat: The automatically-generated message used in `ModuleMessage` has a placeholder body, specifically: `Operation Failed`, which should stand out and be searchable.

File: mln/management/commands/import_mln_xml.py [hunk 2]
```diff
@@ -113,40 +113,42 @@ def handle(self, *args, **options):
...
-				for friend_yield in yield_elem.findAll("friendYield"): 
+				for friend_yield in yield_elem.findall("friendYield/items"): 
 					item_id = int(friend_yield.get("itemID"))
 					qty = int(friend_yield.get("qty"))
 					probability = 100  # not in the XML data
+					# We need to explicitly check for duplicates here because we create a new MessageTemplate otherwise.
+					# There is no other way to detect if a MessageTemplate is being used for a ModuleMessage.
+					if ModuleMessage.objects.filter(module_item_id=id).exists(): continue
 					message_placeholder = MessageTemplate.objects.create(body_id=46304)  # Message body placeholder
 					t[MessageTemplate].append(message_placeholder)
```
Comment (lcdr) [inline mln/management/commands/import_mln_xml.py:150]: Doing both `.objects.create` and `t[MessageTemplate].append` will lead to the same entry being created twice, once by the `.objects.create` and once by the bulk insertion done for `t` later. Since we're actually dealing with a new data model here, and not just importing things with an existing ID from the XML, there's probably no way to use `t` bulk creation since we need the (newly-created) ID to reference it in MessageTemplateAttachment and ModuleMessage. However I'm not entirely sure about this, you could try without the `objects.create` just using `t` and see what happens. If that doesn't work, we'll probably need to use `.objects.create` solely.

Please check if the duplicates were actually introduced by this duplicated creation, maybe we don't need the `.exists` check after all. If we do still need it, could you elaborate on when duplicates can still happen?

Comment (Levi-Lesches) [reply mln/management/commands/import_mln_xml.py]: Thanks, good spot. I got rid of it, but interestingly it wasn't actually creating duplicate entries. Not sure why, since `MessageTemplate` doesn't have a `UniqueConstraint`, so I guess that was some Django optimization. 

There was, however, an error that I somehow missed because I referenced a message body that hasn't been `bulk_create`d yet. I solved that problem by making a new `MessageBody` with an ID of 1 (so it doesn't conflict with the others in the XML) and refer to _that_ instead as the dummy `MessageBody` in a `MessageTemplate`.

> Please check if the duplicates were actually introduced by this duplicated creation, maybe we don't need the `.exists` check after all. If we do still need it, could you elaborate on when duplicates can still happen?

So that's actually two different things. The duplicate line you called out looked like it was creating two `MessageTemplate`s per module _each run of the script_. The duplicate check in the `if` statement is checking if a `ModuleMessage`/`MessageTemplate` pair has already been generated _in a previous run of the script_. The reason this is needed is because there's no way to add a `UniqueConstraint` to a `MessageTemplate` -- this is a case where having many templates with the same body is acceptable. So unlike the other models where `ignore_conflicts=True` was enough to get rid of the explicit check, here we have to do it manually. And because the `MessageTemplate` class isn't explicitly linked to modules, we need to indirectly check if it was generated by checking for the associated `ModuleMessage` instead. 

In other words, if we get rid of that duplicate check, every time you run the script it would generate new `ModuleMessage`s, `MessageTemplate`s, and `MessageTemplateAttachment`s for every `friendYield` in the XML (about 58 in total), which would then result in duplicate rewards being sent out.
